Test/ParallelPrimitives/RadixSortKernels.h

```diff
@@ -573,7 +573,40 @@ extern "C" __global__ void ParallelExclusiveScanSingleWG( int* gCount, int* gHis
 	}
 }
 
-extern "C" __global__ void ParallelExclusiveScanAllWG( int* gCount, int* gHistogram, int* gPartialSum )
+extern "C" __device__ void WorkgroupSync( int threadId, int blockId, int currentSegmentSum, int* currentGlobalOffset, volatile int* gPartialSum, volatile bool* gIsReady )
+{
+	if( threadId == 0 )
+	{
+		int offset = 0;
+
+		if( blockId != 0 )
+		{
+			while( !gIsReady[blockId - 1] )
+			{
+			}
+
+			offset = gPartialSum[blockId - 1];
+
+			__threadfence();
+
+			// Reset the value
+			gIsReady[blockId - 1] = false;
+		}
+
+		gPartialSum[blockId] = offset + currentSegmentSum;
+
+		// Ensure that the gIsReady is only modified after the gPartialSum is written.
+		__threadfence();
+
+		gIsReady[blockId] = true;
+
+		*currentGlobalOffset = offset;
```

Comment:
`currentGlobalOffset` is in shared memory (LDS) and its value will be propagate to all workitems ins a workgroup.